### ISSUE TITLE
Proper way to declare a const int variable?

### ISSUE THREAD

**Issue by sam0x17** [isaacholt100/bnum#17]
I need to declare a `const MY_NUM: I512` but I can't figure out syntax that would allow me to set the value in a const context. Is there not a way to do this yet? Would be happy to write a macro for this purpose if there is any possible way to initialize one of these in a const context.

**Comment by sam0x17** [comment]
That would be awesome. In the mean time is there an example of initializing
one with one of the const methods?

**Comment by isaacholt100** [comment]
I think the simplest example would be using the `from_digits` method on `U512` to convert an array of digits to a `U512` and then using the `from_bits` method on `I512` to convert that `U512` to an `I512`. So for example (I'm using `I256` here for readability), if the hexadecimal literal was `0x5501ece471b59c0ace87fdad70306960eca1623f20207afa6e316e83cb2e3` and you wanted this represented as an `I256`, you could use:

```
use bnum::types::{U256, I256};

const MY_NUM: I256 = {
    let d1 = 0xafa6e316e83cb2e3u64;
    let d2 = 0x960eca1623f20207u64;
    let d3 = 0xc0ace87fdad70306u64;
    let d4 = 0x5501ece471b59u64;
    I256::from_bits(U256::from_digits([d1, d2, d3, d4]))
};
```
The `from_digits` method takes the least significant digit first so the lower order part of the literal first is the first item in the array. Hopefully this makes sense!

**Comment by isaacholt100** [comment]
And then you could use methods like `wrapping_shl`, `rotate_right`, `wrapping_add`, etc. which are `const` to manipulate this further if needed.

**Comment by sam0x17** [comment]
Yes thank you!

**Comment by isaacholt100** [comment]
No worries! Good news, I have just made the `from_str_radix` on unsigned integers `const`, so once I have tested it fully I will publish a new version with this change. I have also created another issue specifically for writing a macro for the purpose you described: #18.

**Comment by isaacholt100** [comment]
I have just published version 0.6.0 which provides a `parse_str_radix` radix method for compile time parsing of integers from string literals, so I will close this issue now.
